src/libraries/System.Console/src/System/ConsolePal.Unix.cs

```diff
@@ -73,14 +73,22 @@ private static SyncTextReader StdInReader
         {
             get
             {
-                EnsureInitialized();
-
-                return Console.EnsureInitialized(
-                        ref s_stdInReader,
-                        () => SyncTextReader.GetSynchronizedTextReader(
-                            new StdInReader(
-                                encoding: Console.InputEncoding,
-                                bufferSize: InteractiveBufferSize)));
+                return Volatile.Read(ref s_stdInReader) ?? EnsureInitialized();
+
+                static SyncTextReader EnsureInitialized()
+                {
+                    EnsureConsoleInitialized();
+
+                    SyncTextReader reader = SyncTextReader.GetSynchronizedTextReader(
+                                                new StdInReader(
+                                                    encoding: Console.InputEncoding,
+                                                    bufferSize: InteractiveBufferSize));
+
+                    // Don't overwrite a set reader.
```

Comment:
StdInReader doesn't store any resources that would need to be cleaned up, right?  Might be worth calling out in a comment.  Up to you.